packages/client-core/src/systems/ui/AvatarDetailView.tsx

```diff
@@ -35,8 +43,30 @@ function createAvatarDetailState(id: string) {
 type CharacterDetailState = ReturnType<typeof createAvatarDetailState>
 
 const CharacterDetailView = () => {
+  const { t } = useTranslation()
   const detailState = useXRUIState() as CharacterDetailState
   const userState = useUserState()
   const user = userState.layerUsers.find((user) => user.id.value === detailState.id.value)
-  return user ? <div style={styles.avatarName as {}}>{user.name.value}</div> : <div></div>
+  const usersTyping = user
+    ? useState(getState(Engine.currentWorld.store, UsersTypingState)[user.id.value]).value
+    : undefined
+
+  useEffect(() => {
+    addActionReceptor(Engine.currentWorld.store, userTypingActionReceptor)
```

Comment:
This will still add a new receptor for every CharacterDetailView, which is not good. We should just add the receptor inside the AvatarUISystem.